### ISSUE TITLE
Evaluation File 

### ISSUE THREAD

**Issue by aliman80** [akshitac8/Generative_MLZSL#10]
1. The dataset argument given in eval.py is not defined.
2. The HYBRID FUSION SELF ATTENTION  module is not defined  in file you need to change it to CLF.
3. When i run the code using given pre-trained weights the accuracies are very low 
4. Without pre trained wieghts accuracies are in range of mAp 20 for zsl

**Comment by akshitac8** [comment]
Hello @aliman80 ,
Thank you for the valuable suggestions. Can you please share your NUS-WIDE training result log file?

**Comment by aliman80** [comment]
[Training_File_Results.pdf](https://github.com/akshitac8/Generative_MLZSL/files/7881046/Training_File_Results.pdf)

**Comment by akshitac8** [comment]
Hi @aliman80 I have attached my log file with the message. In the log file you shared the encoder loss is very high. Please check if all the inputs are passed correctly to the code and also the uploaded code was developed on Tesla v100 machine. So, if you are using a different machine make sure your environment is developed correctly and if you are still not able to get the results you might have to do certain parameter changes to achieve near by results.
[pami_reproduce_git_nus_wide.log](https://github.com/akshitac8/Generative_MLZSL/files/7881374/pami_reproduce_git_nus_wide.log)

**Comment by aliman80** [comment]
Hi,
Thank you very much for your response. I have used the code directly from the repository without any change . Any last comments what inputs specifically should i check and if you have any reference how much those inputs be so that i can compare.


Thanks again

ali
________________________________

**Comment by akshitac8** [comment]
Hi @aliman80 you can check the classifier batch size, learning rate. further you can also check the ratio of seen, unseen labels used for preparing the synthetic data. You can compose the json files of NUS-WIDE dataset using the uploaded features. You can manipulate the size of the dataset in the data loader.

**Comment by aliman80** [comment]
Thank you very much i will do it .

Thanks again
________________________________

**Comment by aliman80** [comment]
One last clarification is that in the case of NUS-WIDE all the features and tags are given so I don't need any preprocessing/extraction . We only need that if we use custom datasets? as mentioned right?
________________________________

**Comment by aliman80** [comment]
As per your suggestion i checked the input features shape,

        train_loc = load_dict_from_hdf5(os.path.join(src, 'nus_wide_vgg_features','nus_seen_train_vgg19.h5'))
        test_unseen_loc = load_dict_from_hdf5(os.path.join(src, 'nus_wide_vgg_features', 'nus_zsl_test_vgg19.h5'))
        test_seen_unseen_loc = load_dict_from_hdf5(os.path.join(src, 'nus_wide_vgg_features', 'nus_gzsl_test_vgg19.h5'))

Here the value of test_unseen_loc vs test_seen_unseen_loc features values are exactly similar in shape , In my understanding they need to be different . So my question is the extracted features which you have provided in google drive for  zsl and gzsl need to be same? i think it should be different. If it makes sense to you and if somehow value of same feature sets is given twice then please update them.


test_unseen_loc.
(107859, 4096)
test_seen_unseen_loc['features'].shape
test_seen_unseen_loc
(107859, 4096)



________________________________

**Comment by aliman80** [comment]
Hi, By using pre trained weights mAP of zsl in range of  3 and GZSL in range of 1 which is quite far away.  Can you update the right weights in pre trained folder please.
